samcli/hook_packages/terraform/hooks/prepare.py

```diff
@@ -905,14 +908,175 @@ def _generate_makefile_rule_for_lambda_resource(
         the terraform project root directory
     python_command_name: str
         the python command name to use for running a script in the makefile rule
+    output_dir: str
+        the directory into which the Makefile is written
 
     Returns
     -------
     str
         The generated makefile rule
     """
-    # TODO
-    return ""
+    target = _get_makefile_build_target(logical_id)
+    resource_address = sam_metadata_resource.resource.get("address", "")
+    apply_command_template = "terraform apply -target {resource_address} -auto-approve"
+    apply_command_recipe = _format_makefile_recipe(apply_command_template.format(resource_address=resource_address))
+    show_command_recipe = _format_makefile_recipe(
+        _build_show_command(
+            python_command_name, output_dir, resource_address, sam_metadata_resource, terraform_application_dir
+        )
+    )
+    return f"{target}{apply_command_recipe}{show_command_recipe}"
```

Comment:
can we formate it a little bit, so it can be a human readable. Something like 
`f"{target}\n\t{apply_command_recipe}\n\t{show_command_recipe}"` and do not forget to consider each OS new line.